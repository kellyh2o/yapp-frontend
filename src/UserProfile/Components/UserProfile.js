import React, { Component } from "react";
import { connect } from "react-redux";

class UserProfile extends Component { 
    render() {
      return (
        <div>
          User Profile
        </div>
      )
    }
}

const mapStateToProps = (state) => {
  return {
    posts: state.posts
  };
}

export default connect(mapStateToProps)(UserProfile);