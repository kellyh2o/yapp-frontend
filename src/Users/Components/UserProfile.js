import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchUsers } from '../Store/users-actions';

class UserProfile extends Component { 
    render() {
      this.props.getUsers();

      return (
        <div>
          User Profile
        </div>
      )
    }
}

const mapStateToProps = (state) => {
  return {
    firstName: state.me.firstName,
    lastName: state.me.lastName,
    email: state.me.email,
    username: state.me.username,
    password: state.me.password,
  };
}

const mapDispatchToProps = (dispatch) => ({
  getUsers: () => {
    dispatch(fetchUsers.request());
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(UserProfile);