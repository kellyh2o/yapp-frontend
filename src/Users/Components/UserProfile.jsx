import React, { useState } from "react";
import { connect } from "react-redux";
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import { UsersActions } from '../Store';

const UserProfile = (props) => { 

  const [firstName, setFirstName] = useState(props.firstName ? props.firstName : "");
  const [lastName, setLastName] = useState(props.lastName ? props.lastName : "");
  const [email, setEmail] = useState(props.email ? props.email : "");
  const [username, setUsername] = useState(props.username ? props.username : "");


  return (
    <div style={{paddingLeft: "20px", width: "400px"}}>
      <Box>
        <Typography variant="h4">
          User Profile
        </Typography>
        <form autoComplete="off" style={{paddingTop: "20px", paddingBottom: "20px"}}>               
            <TextField
              label="First Name"
              variant="outlined"
              type="username"
              defaultValue={firstName}
              onChange={(event) => setFirstName(event.target.value)}
              style={{width: "300px", marginTop: "10px", marginBottom: "10px"}}
            />               
            <TextField
              label="Last Name"
              variant="outlined"
              type="username"
              defaultValue={lastName}
              onChange={(event) => setLastName(event.target.value)}
              style={{width: "300px", marginTop: "10px", marginBottom: "10px"}}
            />               
            <TextField
              label="Email"
              variant="outlined"
              type="username"
              defaultValue={email}
              onChange={(event) => setEmail(event.target.value)}
              style={{width: "300px", marginTop: "10px", marginBottom: "10px"}}
            />               
            <TextField
              label="Username"
              variant="outlined"
              type="username"
              defaultValue={username}
              onChange={(event) => setUsername(event.target.value)}
              style={{width: "300px", marginTop: "10px", marginBottom: "10px"}}
            />
            <Button 
                variant="contained" 
                color="primary"
                disabled={
                  props.firstName === firstName &&
                  props.lastName === lastName && 
                  props.username === username &&
                  props.email === email  
                }
                onClick={() => {
                  props.updateMe(props.token, firstName, lastName, email, username);
                }}
                style={{width: "300px", marginTop: "10px", marginBottom: "10px"}}>
            Save Changes
            </Button>

        </form>

      </Box>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    token: state.token,
    firstName: state.me ? state.me.firstName : null,
    lastName: state.me ? state.me.lastName : null,
    email: state.me ? state.me.email : null,
    username: state.me ? state.me.username : null,
  };
}

const mapDispatchToProps = (dispatch) => ({
  updateMe: (token, firstName, lastName, email, username) => {
    dispatch(UsersActions.updateMe.request(
      {
        token, 
        user: { 
          firstName, 
          lastName, 
          email, 
          username 
        }
      }
    ));
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(UserProfile);