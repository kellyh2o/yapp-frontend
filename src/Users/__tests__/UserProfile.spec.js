import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import UserProfile from "../Components/UserProfile";

describe('UserProfile', () => {
    it("should render", () => {
        const mockStore = configureMockStore();

        const initialState = {
            demoLocation: {
                name: "Hogwarts",
                jumps: [ 
                    {
                        _id: "1",
                        name: "Jump1",
                        events: []
                    },
                    {
                        _id: "2",
                        name: "Jump2",
                        events: []
                    },
                    {
                        _id: "3",
                        name: "Jump3",
                        events: [
                            {
                                _id: "3a",
                                riderMass: 22,
                                riderSpeed: 12,
                            }
                        ]
                    }
                ]
            }
        };
        const store = mockStore(initialState);

        const { asFragment } = render(
            <Provider store={store}>
                <UserProfile />
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    })
});

