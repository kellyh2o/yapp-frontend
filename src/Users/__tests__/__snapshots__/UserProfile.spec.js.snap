// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should render 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
      style="width: 100%;"
    >
      <div
        class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          style="width: auto; margin: auto;"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
          >
            My Account
          </h2>
        </div>
        <form
          autocomplete="off"
          style="padding-top: 20px; padding-bottom: 20px; width: auto; margin: auto;"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="width: 300px; margin-top: 10px; margin-bottom: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              First Name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                type="username"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-221 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-223"
                >
                  <span>
                    First Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="width: 300px; margin-top: 10px; margin-bottom: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Last Name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                type="username"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-221 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-223"
                >
                  <span>
                    Last Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="width: 300px; margin-top: 10px; margin-bottom: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                type="username"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-221 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-223"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
            style="width: 300px; margin-top: 10px; margin-bottom: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Username
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                type="username"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-221 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-223"
                >
                  <span>
                    Username
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            style="width: 300px; margin-top: 10px; margin-bottom: 10px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Update Account
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiBox-root MuiBox-root-258"
            style="padding-top: 20px; width: 300px;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorError MuiTypography-alignCenter"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
